src/Symfony/Bundle/DebugBundle/Resources/config/services.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Bridge\Monolog\Command\ServerLogCommand;
+use Symfony\Bridge\Twig\Extension\DumpExtension;
+use Symfony\Component\HttpKernel\DataCollector\DumpDataCollector;
+use Symfony\Component\HttpKernel\EventListener\DumpListener;
+use Symfony\Component\VarDumper\Cloner\VarCloner;
+use Symfony\Component\VarDumper\Command\Descriptor\CliDescriptor;
+use Symfony\Component\VarDumper\Command\Descriptor\HtmlDescriptor;
+use Symfony\Component\VarDumper\Command\ServerDumpCommand;
+use Symfony\Component\VarDumper\Dumper\CliDumper;
+use Symfony\Component\VarDumper\Dumper\ContextProvider\CliContextProvider;
+use Symfony\Component\VarDumper\Dumper\ContextProvider\RequestContextProvider;
+use Symfony\Component\VarDumper\Dumper\ContextProvider\SourceContextProvider;
+use Symfony\Component\VarDumper\Dumper\ContextualizedDumper;
+use Symfony\Component\VarDumper\Dumper\HtmlDumper;
+use Symfony\Component\VarDumper\Server\Connection;
+use Symfony\Component\VarDumper\Server\DumpServer;
+
+return static function (ContainerConfigurator $container) {
+    $container->parameters()
+        ->set('env(VAR_DUMPER_SERVER)', '127.0.0.1:9912')
+    ;
+
+    $container->services()
+
+        ->set('twig.extension.dump', DumpExtension::class)
+            ->tag('twig.extension')
+            ->args([
+                service('var_dumper.cloner'),
+                service('var_dumper.html_dumper'),
+            ])
+
+        ->set('data_collector.dump', DumpDataCollector::class)
+            ->public()
+            ->tag('data_collector', [
+                'id' => 'dump',
+                'template' => '@Debug/Profiler/dump.html.twig',
+                'priority' => 240,
+            ])
+            ->args([
+                service('debug.stopwatch')->ignoreOnInvalid(),
+                service('debug.file_link_formatter')->ignoreOnInvalid(),
+                param('kernel.charset'),
+                service('request_stack'),
+                null, // var_dumper.cli_dumper or var_dumper.server_connection when debug.dump_destination is set
+            ])
+
+        ->set('debug.dump_listener', DumpListener::class)
+            ->tag('kernel.event_subscriber')
+            ->args([
+                service('var_dumper.cloner'),
+                service('var_dumper.cli_dumper'),
+                null,
+            ])
+
+        ->set('var_dumper.cloner', VarCloner::class)
+            ->public()
+
+        ->set('var_dumper.cli_dumper', CliDumper::class)
+            ->args([
+                null, // debug.dump_destination
```

Comment:
I think this should use `abstract_atg` instead

Reply:
Hi @stof, I was not sure about that. What about the other nullables? Shall we use `abstract_arg(...)` there too:

- https://github.com/symfony/symfony/pull/37237/files#diff-70c3b9dc0cda50bcbb22d93d02621fb3R57
- https://github.com/symfony/symfony/pull/37237/files#diff-70c3b9dc0cda50bcbb22d93d02621fb3R65
- https://github.com/symfony/symfony/pull/37237/files#diff-70c3b9dc0cda50bcbb22d93d02621fb3R93

Reply:
@stof I adjusted all places where the argument is always set by the DI extension.